src/NServiceBus.Core/Serializers/SimpleJson/SimpleJson.g.cs

```diff
@@ -1012,17 +1014,18 @@ static bool SerializeValue(IJsonSerializerStrategy jsonSerializerStrategy, objec
                 success = SerializeString(stringValue, builder);
             else
             {
-                IDictionary<string, object> dict = value as IDictionary<string, object>;
-                if (dict != null)
+                var dic = value as IDictionary;
```

Comment:
we could potentially cover more cases correctly by checking for `IDictionary<T,V>` but this requires quite a bit more code and some effort to get to the keys and values although we're not really interested in the specific types. Maybe one of you has a good idea @andreasohlund @bording ?

Reply:
These are all things that are handled properly by other serializers, say Json.NET?

Reply:
@bording haven't had the chance to verify the behavior in JSON.NET. I think it would be worth to check that though.